src/Symfony/Component/Config/Resource/DirectoryResource.php

```diff
@@ -87,7 +87,12 @@ public function isFresh($timestamp)
                 continue;
             }
 
-            $newestMTime = max($file->getMTime(), $newestMTime);
+            // early return if any file mtime is greater than passed timestamp
+            if (($fileMTime = $file->getMTime()) > $timestamp) {
+                return false;
+            }
+
+            $newestMTime = max($fileMTime, $newestMTime);
```

Comment:
Something like this greatly simplifies the code in this PR, and I think may be a decent option (this diff is against this PR):

```diff
diff --git a/src/Symfony/Component/Config/Resource/DirectoryResource.php b/src/Symfony/Component/Config/Resource/DirectoryResource.php
index 9011cfe..501f8a5 100644
--- a/src/Symfony/Component/Config/Resource/DirectoryResource.php
+++ b/src/Symfony/Component/Config/Resource/DirectoryResource.php
@@ -68,7 +68,7 @@ class DirectoryResource implements ResourceInterface, \Serializable
             return false;
         }
 
-        if ($timestamp < $newestMTime = filemtime($this->resource)) {
+        if ($timestamp <= filemtime($this->resource)) {
             return false;
         }
 
@@ -85,14 +85,12 @@ class DirectoryResource implements ResourceInterface, \Serializable
             }
 
             // early return if a file's mtime exceeds the passed timestamp
-            if ($timestamp < $fileMTime = $file->getMTime()) {
+            if ($timestamp < $file->getMTime()) {
                 return false;
             }
-
-            $newestMTime = max($fileMTime, $newestMTime);
         }
 
-        return $newestMTime < $timestamp;
+        return true;
     }
 
     public function serialize()
```